hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieMetadataTableValidator.java

```diff
@@ -783,6 +788,17 @@ private void validateAllFileGroups(
           .flatMap(HoodieFileGroup::getAllFileSlices).sorted(new FileSliceComparator())
           .collect(Collectors.toList());
     }
+    if (!StringUtils.isNullOrEmpty(cfg.validateLastNFileSlices)) {
+      int lastNFileSlices = Integer.parseInt(cfg.validateLastNFileSlices);
+      int allFileSlicesFromFSCount = allFileSlicesFromFS.size();
+      if (allFileSlicesFromFSCount > lastNFileSlices) {
+        allFileSlicesFromFS = allFileSlicesFromFS.subList(allFileSlicesFromFSCount - lastNFileSlices, allFileSlicesFromFSCount);
```

Comment:
allFielSlicesFromFS is already sorted by commit time right?

Reply:
HoodieFileGroup maintains a treemap of file slices internally and hence HoodieFilegroup.getAllFileSlices is already sorted based on commit time.